### PR TITLE
Convert `Hero.md` and `contribution-flow` to Storybook

### PR DESCRIPTION
**Pull request by SudharakaP**
Some partly done work I did yesterday to convert `Hero.md` and `contribution-flow` to Storybook. Will update as I go along. 😄 🐨 

Related to https://github.com/opencollective/opencollective/issues/4313

### CODE REVIEW

**Comment by SudharakaP** [comment]
@Betree : I've added support to view the `contribution-flow` and `Hero.js` components on Storybook. It seemed to me that the `contribution-flow` elements in `Styleguidist` were quite outdated version; so I tried to make the Storybook version based on the new versions of these components. However I am not sure exactly if we should be adding more interactions in Storybook for these components. Anyways, let me know your thoughts. 🤔

File: stories/collective-page/Hero.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  component={Hero}
+  argTypes={{
+    collective: { defaultValue: webpackCollective },
+    host: { defaultValue: openSourceHost },
+  }}
+  parameters={{
+    actions: {
+      handles: ['mouseover', 'click'],
+    },
+  }}
```
Comment (Betree) [inline stories/collective-page/Hero.stories.mdx:16]: Unless I'm missing something, I don't think logging these events is really interesting, is it?

```suggestion
```

File: stories/contribution-flow/StepDetails.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { flexibleIntervalTier, recurringIntervalTier } from '../mocks/tiers';
+
+<Meta
+  title="Contribution Flow/StepDetails"
+  component={StepDetails}
+  parameters={{
+    actions: {
+      handles: ['mouseover', 'click'],
+    },
+  }}
```
Comment (Betree) [inline stories/contribution-flow/StepDetails.stories.mdx:13]:
```suggestion
```

File: stories/contribution-flow/StepPayment.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const stripePromise = loadStripe('storybookStripeId');
+
+<Meta
+  title="Contribution Flow/StepPayment"
+  component={StepPayment}
+  parameters={{
+    actions: {
+      handles: ['mouseover', 'click'],
+    },
+  }}
```
Comment (Betree) [inline stories/contribution-flow/StepPayment.stories.mdx:18]:
```suggestion
```

File: stories/contribution-flow/StepProfile.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { webpackCollective } from '../mocks/collectives';
+
+<Meta
+  title="Contribution Flow/StepProfile"
+  component={StepProfile}
+  parameters={{
+    actions: {
+      handles: ['mouseover', 'click'],
+    },
+  }}
```
Comment (Betree) [inline stories/contribution-flow/StepProfile.stories.mdx:13]:
```suggestion
```

**Review by Betree — COMMENTED** [review]
Looks great as a first version, the `Hero` story is especially useful and very easy to customize :tada: 

Only blocker on the change for `components/contribution-flow/StepPayment.js`, I did not understand this change

File: components/contribution-flow/StepPayment.js [hunk 5]
```diff
@@ -164,7 +164,7 @@ const StepPayment = ({
                   )}
                 </Flex>
               </Flex>
-              {value.key === NEW_CREDIT_CARD_KEY && checked && (
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/contribution-flow/StepPayment.js]: @Betree : Ops, I was testing something and forgot to re-add this back. Sorry about that. I've correct this. 👍🏽

File: stories/collective-page/Hero.stories.mdx (hunk 1 above)
Comment (SudharakaP) [reply stories/collective-page/Hero.stories.mdx]: Yes we could remove these. I've removed all of those. 😺

**Review by Betree — APPROVED** [review]
:+1:

**Pull request merged**